modules/tinymce/src/core/main/ts/autocomplete/AutocompleteContext.ts

```diff
@@ -37,10 +37,15 @@ const findStart = (dom: DOMUtils, initRange: Range, ch: string, minChars: number
     return Optional.none();
   }
 
-  const findTriggerChIndex = (element: Text, offset: number, text: string) =>
+  const buf: Array<{text: string; offset: number}> = [];
```

Comment:
```suggestion
  const buf: Array<{ text: string; offset: number }> = [];
```

Reply:
Also can you please add a comment about what's happening here, as it's not very clear without looking at it in more detail.

Reply:
Actually having looked into this more... if you're doing mutation anyways then having to do the `Arr.foldl` below is rather inefficient and we might as well just update as we go. For example:
```ts
const buffer = { text: '', offset: 0 };

const findTriggerChIndex = (element: Text, offset: number, text: string) => {
  buffer.text = text + buffer.text;
  buffer.offset += offset;
  // Stop searching by just returning the current offset if whitespace was found (eg Optional.none())
  // and we'll handle the final checks below instead
  return findTrigger(buffer.text, buffer.offset, ch).getOr(offset);
};
```

- [ ] Don't loop over the buffer to create the text/offsets on every text node when it's not needed